nsxt/resource_nsxt_policy_ipsec_vpn_service.go

```diff
@@ -0,0 +1,324 @@
+		}
+		var destList []string
+		for _, dest := range rule.Destinations {
+			destList = append(destList, *dest.Subnet)
+		}
+		nsxID := rule.Id
+		elem["nsx_id"] = nsxID
+		elem["sources"] = srcList
+		elem["destinations"] = destList
+		elem["action"] = rule.Action
+		ruleList = append(ruleList, elem)
+	}
+	err := d.Set("bypass_rules", ruleList)
+	if err != nil {
+		log.Printf("[WARNING] Failed to set bypass_rules in schema: %v", err)
+	}
+}
+
+func getIPSecVPNBypassRulesFromSchema(d *schema.ResourceData) []model.IPSecVpnRule {
+	rules := d.Get("bypass_rules")
+	if rules != nil {
+		rules := rules.([]interface{})
+		var ruleList []model.IPSecVpnRule
+		for _, rule := range rules {
+			data := rule.(map[string]interface{})
+			action := data["action"].(string)
+			sourceRanges := interface2StringList(data["sources"].(*schema.Set).List())
+			destinationRanges := interface2StringList(data["destinations"].(*schema.Set).List())
+
+			/// Source Subnets
+
+			SourceIPSecVpnSubnetList := make([]model.IPSecVpnSubnet, 0)
+			if len(sourceRanges) > 0 {
+				for _, element := range sourceRanges {
+					subnet := element
+					IPSecVpnSubnet := model.IPSecVpnSubnet{
+						Subnet: &subnet,
+					}
+					SourceIPSecVpnSubnetList = append(SourceIPSecVpnSubnetList, IPSecVpnSubnet)
+				}
+			}
+
+			/// Destination Subnets
+			DestinationIPSecVpnSubnetList := make([]model.IPSecVpnSubnet, 0)
+			if len(destinationRanges) > 0 {
+				for _, element := range destinationRanges {
+					subnet := element
+					IPSecVpnSubnet := model.IPSecVpnSubnet{
+						Subnet: &subnet,
+					}
+					DestinationIPSecVpnSubnetList = append(DestinationIPSecVpnSubnetList, IPSecVpnSubnet)
+				}
+			}
+			ruleID := data["nsx_id"].(string)
+			if ruleID == "" {
+				ruleID = newUUID()
+			}
+			elem := model.IPSecVpnRule{
+				Action:       &action,
+				Sources:      SourceIPSecVpnSubnetList,
+				Destinations: DestinationIPSecVpnSubnetList,
+				UniqueId:     &ruleID,
+				Id:           &ruleID,
+			}
+			ruleList = append(ruleList, elem)
+		}
+		return ruleList
+	}
+	return nil
+}
+
+func resourceNsxtPolicyIPSecVpnServiceRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining IPSecVpnService ID")
+	}
+	localeServicePath := d.Get("locale_service_path").(string)
+	isT0, gwID, localeServiceID, err := parseLocaleServicePolicyPath(localeServicePath)
+	if err != nil {
+		return err
+	}
+	obj, err := getNsxtPolicyIPSecVpnServiceByID(connector, gwID, isT0, localeServiceID, id, isPolicyGlobalManager(m))
+	if err != nil {
+		return handleReadError(d, "IPSecVpnService", id, err)
+	}
+	d.Set("display_name", obj.DisplayName)
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+	d.Set("enabled", obj.Enabled)
+	d.Set("ha_sync", obj.HaSync)
+
+	if obj.BypassRules != nil {
```

Comment:
if rules are deleted, we need to make sure they are deleted in schema, so I don't think this `if` is needed?